subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformWithDependenciesIntegrationTest.groovy

```diff
@@ -325,4 +329,220 @@ project(':app') {
         outputLines.any { it ==~ /TestTransform: .*slf4j-api-1.7.25.jar is not up-to-date because:/ }
         outputLines.any { it == "Single step transform received dependencies files [slf4j-api-1.7.25.jar, common.jar] for processing lib.jar" }
     }
+
+    def "transforms with different dependencies in multiple dependency graphs are executed"() {
+        given:
+        mavenHttpRepo.module("org.slf4j", "slf4j-api", "1.7.26").publish().allowAll()
+        settingsFile << "include('app2')"
+        buildFile << """
+            project(':app') {
+                task resolveSize(type: Copy) {
+                    def artifacts = configurations.implementation.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'size') }
+                    }.artifacts
+                    from artifacts.artifactFiles
+                    into "\${buildDir}/libs/size"
+                }
+                task resolveInter(type: Copy) {
+                    def artifacts = configurations.implementation.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'inter') }
+                    }.artifacts
+                    from artifacts.artifactFiles
+                    into "\${buildDir}/libs/inter"
+                }         
```

Comment:
I'm not sure what `inter` is supposed to mean here. It would probably be easier to understand if we used a full word at least.